pkg/remotecmd/kubeexec.go

```diff
@@ -0,0 +1,294 @@
+
+	"github.com/redhat-developer/odo/pkg/devfile/adapters/common"
+	"github.com/redhat-developer/odo/pkg/kclient"
+	"github.com/redhat-developer/odo/pkg/storage"
+	"github.com/redhat-developer/odo/pkg/task"
+	"github.com/redhat-developer/odo/pkg/util"
+)
+
+// kubeExecProcessHandler implements RemoteProcessHandler by executing Devfile commands right away in the container
+// (like a 'kubectl exec'-like approach). Command execution is done in the background, in a separate goroutine.
+// It works by storing the parent process PID in a file (_startCmdProcessPidFile) in the container,
+// then fires the exec command in the background.
+// The goroutine started can then be stopped by killing the process stored in the state file (_startCmdProcessPidFile)
+// in the container.
+type kubeExecProcessHandler struct{}
+
+//This allows to verify interface compliance at compile-time.
+//See https://github.com/redhat-developer/odo/wiki/Dev:-Coding-Conventions#verify-interface-compliance
+var _ RemoteProcessHandler = (*kubeExecProcessHandler)(nil)
+
+func NewKubeExecProcessHandler() *kubeExecProcessHandler {
+	return &kubeExecProcessHandler{}
+}
+
+// GetProcessInfoForCommand returns information about the process representing the given Devfile command.
+// A PID of 0 denotes a Stopped process.
+func (k *kubeExecProcessHandler) GetProcessInfoForCommand(
+	devfileCmd devfilev1.Command,
+	kclient kclient.ClientInterface,
+	podName string,
+	containerName string,
+) (RemoteProcessInfo, error) {
+	klog.V(4).Infof("GetProcessInfoForCommand for %q", devfileCmd.Id)
+
+	pid, err := getRemoteProcessPID(kclient, devfileCmd, podName, containerName)
+	if err != nil {
+		return RemoteProcessInfo{}, err
+	}
+
+	return k.getProcessInfoFromPid(pid, kclient, podName, containerName)
+}
+
+// StartProcessForCommand runs the (potentially never finishing) Devfile command in the background.
+// The goroutine spawned here can get stopped either by stopping the parent process (e.g., 'odo dev'),
+// or by stopping the underlying remote process by calling the StopProcessForCommand method.
+func (k *kubeExecProcessHandler) StartProcessForCommand(
+	devfileCmd devfilev1.Command,
+	kclient kclient.ClientInterface,
+	podName string,
+	containerName string,
+	outputHandler CommandOutputHandler,
+) error {
+	klog.V(4).Infof("StartProcessForCommand for %q", devfileCmd.Id)
+
+	if devfileCmd.Exec == nil {
+		return errors.New(" only Exec commands are supported")
+	}
+
+	// deal with environment variables
+	cmdLine := devfileCmd.Exec.CommandLine
+	setEnvVariable := util.GetCommandStringFromEnvs(devfileCmd.Exec.Env)
+	if setEnvVariable != "" {
+		cmdLine = setEnvVariable + " && " + devfileCmd.Exec.CommandLine
+	}
+
+	// Change to the workdir and execute the command
+	pidFile := getPidFileForCommand(devfileCmd)
+	cmd := []string{common.ShellExecutable, "-c"}
+	// Storing the /bin/sh parent process PID. It will allow to determine its children later on and kill them when a stop is request
```

Comment:
```suggestion
	// Storing the /bin/sh parent process PID. It will allow to determine its children later on and kill them when a stop is requested
```